### ISSUE TITLE
Nested tempdir example

### ISSUE THREAD

**Issue by Masynchin** [Stebalien/tempfile#190]
I am struggling to find how to create nested temporary directory. The documentation has no examples about it. I think that appropriate place for such example is the [`tempdir_in`](https://github.com/Stebalien/tempfile/blob/ef7ffc0e882771a4b872305dc0a29cc0ce3e3a61/src/dir.rs#L113)'s docs. Also, it seems that documentation for this function is copy-pasted from `tempdir`.

**Comment by Stebalien** [comment]
`tempdir` was merged into this crate, so that makes sense.

Creating a nested temporary directory is a matter of:

```rust
let tmp_a = TempDir::new()?;
let tmp_b = TempDir::new_in(&tmp_a)?;
```

What was your confusion with the docs?

**Comment by Masynchin** [comment]
> `tempdir` was merged into this crate, so that makes sense.

I meant [`tempdir`](https://github.com/Stebalien/tempfile/blob/ef7ffc0e882771a4b872305dc0a29cc0ce3e3a61/src/dir.rs#L64) function (not the crate), from which [tempdir_in](https://github.com/Stebalien/tempfile/blob/ef7ffc0e882771a4b872305dc0a29cc0ce3e3a61/src/dir.rs#L113)'s documentation is copy-pasted. So, currently there are two functions with exact same docs.

**Comment by Stebalien** [comment]
Ah. Hm. Yes, you're right. Fix in #191. LGTY?

**Comment by Masynchin** [comment]
> Fix in #191. LGTY?

Yes, but it provides only example with real FS path (`"."`), and not with `TempDir`. I think this is not clear that it can also accept `TempDir`, since it also implements `AsRef<Path>` trait.

**Comment by Stebalien** [comment]
The user can also just call `tempdir_in(tmp_a.path())`. Examples augment the documentation but aren't (and can't be) exhaustive.
